pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -887,7 +889,14 @@ func (p *Provisioner) restoreClusterBackup(cluster *v3.Cluster, spec v3.ClusterS
 		return "", "", "", fmt.Errorf("snapshot [%s] is not a backup of cluster [%s]", backup.Name, cluster.Name)
 	}
 
-	api, token, cert, err = p.driverRestore(cluster, spec)
+	if filename, err := GetBackupFilenameFromURL(backup.Spec.Filename); err == nil { // s3 file
+		// need to remove extention
+		snapshot = strings.TrimSuffix(filename, path.Ext(filename))
+	} else if len(backup.Spec.Filename) != 0 { // not s3 url
```

Comment:
Can you add a comment when this situation is hit? AFAIK only backups that have filename set is s3 with an URL, for local its not set. Is this to catch filenames with no URL, if so, what case is that hit?

Reply:
There are 3 cases here:
1 - backup is s3
2 - backup is local (contains timestamp in the filename)
3 - backup is local (created with a previous version of rancher where filename is not used)

This condition handles case number 2.